[100%] Testing mps writer

a-zakir opened the pull request.

a-zakir committed:
add tests

flomnes commented on src/tests/src/io/outputs/testMpsWriter.cpp:
```diff
@@ -0,0 +1,158 @@
+        for (int vi = 0; vi < fromMps.num_variables(); ++vi)
+        {
+            isLp &= !fromMps.VarIsIntegral(vi);
+        }
```
std::all_of

a-zakir replied:
yes that would be the right way if i had a vector of variable fromMps

flomnes commented on src/tests/src/io/outputs/testMpsWriter.cpp:
```diff
@@ -0,0 +1,158 @@
+            BOOST_CHECK(origVariable->getName() == fromMps.VarName(vi));
+            BOOST_CHECK(origVariable->isInteger() == fromMps.VarIsIntegral(vi));
+            BOOST_CHECK(origVariable->getLb() == fromMps.VarLowerBound(vi));
+            BOOST_CHECK(origVariable->getUb() == fromMps.VarUpperBound(vi));
```
BOOST_CHECK_EQUAL

flomnes commented on src/tests/src/io/outputs/testMpsWriter.cpp:
```diff
@@ -0,0 +1,158 @@
+                run(subEntry.path());
+            }
+        }
+        run(entry.path());
```
Not sure I understand why there are two distinct calls to `run`

a-zakir replied:
Because the studies in src/tests/resources/modeler/epic2/us2.5/ are not at the same level as those in src/tests/resources/modeler/.

flomnes commented on src/tests/src/io/outputs/testMpsWriter.cpp:
```diff
@@ -0,0 +1,158 @@
+
+        checkProblem(modeler.masterProblem(), studyPath / "output" / "master.mps");
+        checkProblem(modeler.subproblems().at(0), studyPath / "output" / "1-1.mps");
+    }
```
IMO this method should be split
1. Create stuff
2. Check stuff

"run" is a bit confusing

a-zakir replied:
that would be possible if `modeler.masterProblem()` and `modeler.subproblems()` are stored...

a-zakir replied:
I can rename this method to checkProblems, in fact, I'll introduce a parameter that disables resolution

a-zakir replied:
ok without further modifications (like moving problems from modeler), i put build lines in separate func.

a-zakir committed:
update

a-zakir committed:
upd

a-zakir committed:
up

a-zakir committed:
upd

a-zakir committed:
compile ok

a-zakir committed:
update namespaces

a-zakir committed:
fix tests

a-zakir committed:
Merge branch 'feature/mps-writer' into feature/test-mps-writer

a-zakir committed:
fix tests

a-zakir committed:
fix tests

a-zakir committed:
update

flomnes committed:
Fix sonar.projectVersion (#3308)

9.3.2 -> 9.3.3

a-zakir committed:
fix mps (#3309)

flomnes committed:
Single problem getter API - fix naming for subproblems (#3312)

flomnes committed:
Single problem getter API - fix day numbers (#3311)

There was a minor bug for daily binding constraints.

## Before
```
 E  Storing_balance::daily::day<0>
 E  Storing_balance::daily::day<0>
 E  Storing_balance::daily::day<0>
 E  Storing_balance::daily::day<0>
 E  Storing_balance::daily::day<0>
 E  Storing_balance::daily::day<0>
 E  Storing_balance::daily::day<0>
```
## After
```
 E  Storing_balance::daily::day<0>
 E  Storing_balance::daily::day<1>
 E  Storing_balance::daily::day<2>
 E  Storing_balance::daily::day<3>
 E  Storing_balance::daily::day<4>
 E  Storing_balance::daily::day<5>
 E  Storing_balance::daily::day<6>
```

a-zakir committed:
update

a-zakir committed:
Merge branch 'develop' into feature/test-mps-writer

a-zakir committed:
safe names

a-zakir committed:
fix

a-zakir committed:
fix test

a-zakir committed:
upd

a-zakir committed:
upd

a-zakir committed:
upd

a-zakir committed:
update test

a-zakir committed:
update test

a-zakir committed:
update test

a-zakir committed:
update test

a-zakir committed:
fix: update Modeler

a-zakir committed:
fix: update Modeler

a-zakir committed:
perf

a-zakir committed:
update

a-zakir committed:
fix: handle null problem in isProblemEmpty

a-zakir committed:
update test: add unit tests for MakeMpsSafeUniqueName and NameManager

a-zakir committed:
fix: clean up formatting and remove unused parameters

a-zakir committed:
format

a-zakir committed:
update

a-zakir committed:
update

a-zakir committed:
fix: replace direct comparisons with utility functions for numerical stability

a-zakir committed:
update

a-zakir committed:
update
